autocheck/plugin.py

```diff
@@ -0,0 +1,212 @@
+    session.hive_mock.__enter__()
+    if settings.setup_hive_mock:
+        session.hive_mock = setup_hive_mock()
+        session.hive_mock.__enter__()
+
+    session.tests_repo = empty_context_manager()
+    session.tests_repo.__enter__()
+
+    session.directory_context = empty_context_manager()
+    session.directory_context.__enter__()
+
+    if settings.clone_segel_repo:
+        session.tests_repo = tempfile.TemporaryDirectory(suffix="___test_files")
+        tests_repo: Path = Path(session.tests_repo.__enter__())
+        clone_tests_repository(
+            settings.segel_gitlab_host,
+            settings.segel_gitlab_token,
+            settings.tests_repository_url,
+            settings.tests_repository_ref,
+            tests_repo,
+        )
+
+        session.directory_context = push_dir(tests_repo)
+        session.directory_context.__enter__()
+
+        # Change the import path to be our new temp dir
+        session.config.args[:] = [str(tests_repo)]
+
+        # Make pytest truly believe it was initialy run from our temp dir (otherwise things look funny or don't work properly)
+        session.path = tests_repo
+        session.config._rootpath = tests_repo
+        session.config.pluginmanager._confcutdir = tests_repo
+        session.config.invocation_params = pytest.Config.InvocationParams(
+            dir=tests_repo, args=session.config.args, plugins=session.config.invocation_params.plugins
+        )
+    
+    session.copied_files = [
+        copy_file(src=autocheck.default_tests.__file__, dst="default_tests.py"),
+    ]
+    for file in session.copied_files:
+        file.__enter__()
+
+
+# Copied from pytest source code and edited to always approve specific files
+@pytest.hookimpl
+def pytest_collect_file(file_path: Path, parent: nodes.Collector) -> Module | None:
+    ALWAYS_COLLECT: list[str] = [
+        "default_tests.py",
+    ]
+
+    if file_path.name in ALWAYS_COLLECT:
+        ihook = parent.session.gethookproxy(file_path)
+        module: Module = ihook.pytest_pycollect_makemodule(
+            module_path=file_path, parent=parent
+        )
+        return module
+    return None
+
+
+@pytest.hookimpl()
+def pytest_sessionfinish(session: pytest.Session, exitstatus: int) -> None:
+    with contextlib.ExitStack() as exit_stack:
+        exit_stack.push(session.hive_mock)
+        exit_stack.push(session.tests_repo)
+        exit_stack.push(session.directory_context)
+        for file in session.copied_files:
+            exit_stack.push(file)
+
+
+def get_all_subpath_options(path_parts: list[str]) -> set[str]:
+    previous_subpath: str = ""
+    options: set[str] = set()
+    for part in path_parts:
+        subpath = previous_subpath + "/" + part
+        subpath = subpath.strip("/")
+        previous_subpath = subpath
+        options.add(subpath)
+
+    return options
+
+
+@pytest.hookimpl()
+def pytest_collection_modifyitems(session: pytest.Session, config: pytest.Config, items):
```

Comment:
Also missing some types

```suggestion
def pytest_collection_modifyitems(session: pytest.Session, _: pytest.Config, items):
```

Reply:
this must be called config because that's how pytest hooks work

Reply:
yeah, probably somthing in pytest's implementation of hooks... maybe it's something like fixtures